### DISCUSSION TITLE
Help for this issue to understand some things that i need to do

### DISCUSSION THREAD

**Discussion by DoTheBestToGetTheBest** [paradigmxyz/reth#4877] [Help] [answered]
Hello everyone and thank you for reading this discussion asking for help.

I'am really loving working in reth, i hope getting better in future in reth.

So i'm taking this issue : https://github.com/paradigmxyz/reth/issues/4874

And what i undertand i need to add Terminate and Terminated in the enum Discv4Command 

What does terminate and terminated should do exactly ? does terminate should stop the discv4 service ? and terminated should verify if it's stopped ?

And also what does Terminate and Terminated should take in argument ? 

Thank you all for your help !

**Comment by mattsse** [comment, marked as answer]
the terminate command variant needs to be sent like this:

https://github.com/paradigmxyz/reth/blob/858ea41da1f8852e93e0604dbb2f7a89a5398b28/crates/net/discv4/src/lib.rs#L366-L375

see also this for reference

https://github.com/paradigmxyz/reth/blob/858ea41da1f8852e93e0604dbb2f7a89a5398b28/crates/net/discv4/src/lib.rs#L323-L329

then it needs to be handled like this:

https://github.com/paradigmxyz/reth/blob/858ea41da1f8852e93e0604dbb2f7a89a5398b28/crates/net/discv4/src/lib.rs#L1532-L1534

* emit a `Terminated` event and return `Poll::Ready` to terminate the future

**Reply by DoTheBestToGetTheBest** [reply]
Amazing, thank you so much !

**Reply by DoTheBestToGetTheBest** [reply]
Currently what i've done : 

creating a Terminated elements in the Discv4Command enums

creating  a service_send for Discv4Command::Terminated elements 
```rust
pub fn terminate(&self) {
        let cmd = Discv4Command::Terminated;
        self.send_to_service(cmd);
    }

``` 

in the Discv4 struct


i created also a match for Discv4command::Terminated => todo! 


What should exactly i do now please?

**Reply by mattsse** [reply]
please open a draft

**Reply by DoTheBestToGetTheBest** [reply]
Alright, thank you for your answer Mat!
